File: src/main/java/net/thedragonteam/thedragonlib/blocks/ItemBlockBasic.java
package net.thedragonteam.thedragonlib.blocks;

import net.minecraft.block.Block;
import net.minecraft.item.ItemStack;
import net.thedragonteam.thedragonlib.config.FeatureWrapper;

import javax.annotation.Nonnull;

public class ItemBlockBasic extends ItemBlockTDLib {

    private FeatureWrapper feature;

    public ItemBlockBasic(Block block) {
        super(block);
    }

    public ItemBlockBasic(Block block, FeatureWrapper feature) {
        this(block);
        this.feature = feature;
        this.setHasSubtypes(feature.variantMap().length > 0);
    }

    @Nonnull
    @Override
    public String getUnlocalizedName(ItemStack stack) {
        return getHasSubtypes() && feature.nameMap.containsKey(stack.getItemDamage()) ? (super.getUnlocalizedName(stack) + "." + feature.nameMap.get(stack.getItemDamage())).replaceAll("=", ".").replaceAll(",", ".") : super.getUnlocalizedName(stack);
    }
}

File: src/main/java/net/thedragonteam/thedragonlib/blocks/ItemBlockBasic.kt
package net.thedragonteam.thedragonlib.blocks

import net.minecraft.block.Block
import net.minecraft.item.ItemStack
import net.thedragonteam.thedragonlib.config.FeatureWrapper

class ItemBlockBasic(block: Block) : ItemBlockTDLib(block) {

    private lateinit var feature: FeatureWrapper

    constructor(block: Block, feature: FeatureWrapper) : this(block) {
        this.feature = feature
        this.setHasSubtypes(feature.variantMap().isNotEmpty())
    }

    override fun getUnlocalizedName(stack: ItemStack?): String {
        return if (getHasSubtypes() && feature.nameMap.containsKey(stack!!.itemDamage)) (super.getUnlocalizedName(stack) + "." + feature.nameMap[stack.itemDamage]).replace("=".toRegex(), ".").replace(",".toRegex(), ".") else super.getUnlocalizedName(stack)
    }
}